sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2960,7 +2968,8 @@ def set_blob_tags(self, tags=None, **kwargs):
         :returns: Blob-updated property dict (Etag and last modified)
         :rtype: Dict[str, Any]
         """
-        options = self._set_blob_tags_options(tags=tags, **kwargs)
+        version_id = kwargs.pop('version_id', None) or self.version_id
```

Comment:
Should just handle this inside `_set_blob_tags_options` rather than here and passing it in.